src/main/java/io/appform/dropwizard/sharding/dao/operations/RunInSession.java

```diff
@@ -0,0 +1,35 @@
+package io.appform.dropwizard.sharding.dao.operations;
+
+import java.util.function.Function;
+import lombok.Data;
+import lombok.NonNull;
+import lombok.experimental.SuperBuilder;
+import org.hibernate.Session;
+
+/**
+ * Run any action within a session on specific shard.
+ *
+ * @param <T> return type of the given action.
+ */
+@Data
+@SuperBuilder
+public class RunInSession<T> extends OpContext<T> {
+
+  @NonNull
+  Function<Session, T> handler;
+
+  @Override
+  public T apply(Session session) {
+    return handler.apply(session);
+  }
+
+  @Override
+  public @NonNull OpType getOpType() {
```

Comment:
is @NonNull required here as we are returning a constant value

Reply:
Removed.